src/main/java/net/minecraftforge/client/model/generators/ModelBuilder.java

```diff
@@ -110,8 +149,14 @@ public JsonObject serialize() {
 
         if (!this.textures.isEmpty()) {
             JsonObject textures = new JsonObject();
-            for (Entry<String, String> e : this.textures.entrySet()) {
-                textures.addProperty(e.getKey(), e.getValue());
+            for (Entry<String, ResourceLocation> e : this.textures.entrySet()) {
+                String texName;
+                if (e.getValue().getNamespace().equals(owningNamespace)) {
```

Comment:
This is not right, inferred domain is "minecraft" not the current domain of the model.